pkg/sources/github/github.go

```diff
@@ -369,38 +305,32 @@ func (s *Source) Chunks(ctx context.Context, chunksChan chan *sources.Chunk, tar
 	githubSecondsSpentRateLimited.WithLabelValues(s.name).Set(0)
 	githubReposScanned.WithLabelValues(s.name).Set(0)
 
-	installationClient, err := s.enumerate(ctx, apiEndpoint)
+	err := s.enumerate(ctx)
 	if err != nil {
-		return err
+		return fmt.Errorf("error enumerating: %w", err)
 	}
 
-	return s.scan(ctx, installationClient, chunksChan)
+	return s.scan(ctx, chunksChan)
 }
 
-func (s *Source) enumerate(ctx context.Context, apiEndpoint string) (*github.Client, error) {
-	var (
-		installationClient *github.Client
-		err                error
-	)
-
-	switch cred := s.conn.GetCredential().(type) {
-	case *sourcespb.GitHub_BasicAuth:
-		if err = s.enumerateBasicAuth(ctx, apiEndpoint, cred.BasicAuth); err != nil {
-			return nil, err
+func (s *Source) enumerate(ctx context.Context) error {
+	// I'm not wild about switching on the connector type here (as opposed to dispatching to the connector itself) but
+	// this felt like a compromise that allowed me to isolate connection logic without rewriting the entire source.
```

Comment:
I agree with this comment. Currently enumeration is different based on how we access GitHub, but I wonder if in the future we could always enumerate the same way and ignore any errors. Maybe that's a bit wasteful with rate-limits though.. 🤔